libs/langgraph/langgraph/pregel/loop.py

```diff
@@ -294,29 +296,17 @@ def put_writes(self, task_id: str, writes: Sequence[tuple[str, Any]]) -> None:
         """Put writes for a task, to be read by the next tick."""
         if not writes:
             return
+        checkpoint_during = self.checkpoint_during or any(w[0] == TASKS for w in writes)
```

Comment:
Could we have a comment *why* we're checkpointing if there are tasks? I'm assuming this is to handle the fact that tasks are only saved to the writes() in the parent checkpoint?